enclave/core/strtoul.c

```diff
@@ -0,0 +1,190 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <openenclave/bits/defs.h>
+#include <openenclave/bits/types.h>
+#include <openenclave/corelibc/ctype.h>
+
+//
+// If c is a digit character:
+//     then: _digit[c] yields the integer value for that digit character.
+//     else: _digit[c] yields 0xFF.
+//
+// Digit characters fall within these ranges: ['0'-'9'] and ['A'-'Z'].
+//
+// Examples:
+//     _digit['9'] => 9
+//     _digit['A'] => 10
+//     _digit['Z'] => 35
+//     _digit['?'] => 0xFF
+//
+static const unsigned char _digit[256] = {
```

Comment:
Right now, it seems there is a bit of duplication and it is not quite obvious why we can't include the MUSL implementation in the static library rather than writing the code here.

Reply:
Yes, there is replication of functionality that is implemented in MUSL and only implemented in oelibc rather than core, for largely the same reasons as discussed in ctype.h unfortunately.

Specifically, `strtoul` is one of the new functions added to support mbedtls and sits on top of that chunk of character manipulation code we should have a more detailed discussion on whether we want to expend the effort to use directly from MUSL. I'd take that as a separate PR.